acme/tls_sni_challenge.go

```diff
@@ -40,12 +40,12 @@ func (t *tlsSNIChallenge) Solve(chlng challenge, domain string) error {
 	return t.validate(t.jws, domain, chlng.URI, challenge{Resource: "challenge", Type: chlng.Type, Token: chlng.Token, KeyAuthorization: keyAuth})
 }
 
-// TLSSNI01ChallengeCert returns a certificate for the `tls-sni-01` challenge
-func TLSSNI01ChallengeCert(keyAuth string) (tls.Certificate, error) {
+// TLSSNI01ChallengeCert returns a certificate and target domain for the `tls-sni-01` challenge
+func TLSSNI01ChallengeCertDomain(keyAuth string) (tls.Certificate, string, error) {
```

Comment:
Yeah, that's what I think would be good. I can see the value in it; it's up
to @xenolf what he wants to do about compatibility. Personally I would
rather break it now than later.

On Sun, Jun 12, 2016, 10:03 AM Derek McGowan <notifications@github.com>
wrote:

> In acme/tls_sni_challenge.go
> <https://github.com/xenolf/lego/pull/227#discussion_r66723191>:
>
> > @@ -40,12 +40,12 @@ func (t *tlsSNIChallenge) Solve(chlng challenge, domain string) error {
> >  	return t.validate(t.jws, domain, chlng.URI, challenge{Resource: "challenge", Type: chlng.Type, Token: chlng.Token, KeyAuthorization: keyAuth})
> >  }
> >
> > -// TLSSNI01ChallengeCert returns a certificate for the `tls-sni-01` challenge
> > -func TLSSNI01ChallengeCert(keyAuth string) (tls.Certificate, error) {
> > +// TLSSNI01ChallengeCert returns a certificate and target domain for the `tls-sni-01` challenge
> > +func TLSSNI01ChallengeCertDomain(keyAuth string) (tls.Certificate, string, error) {
>
> To be clear, you are suggesting just updating the original function to now
> return the domain? I am OK doing that but was unsure the impact it would
> have on importers since it is already an exported function.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/xenolf/lego/pull/227/files/c8b0781028a221cd4dfebaa0f08f28e02541817c#r66723191>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABE5kbmPCRktWfA7TbTRFCpcMlY8k_n-ks5qLC3LgaJpZM4IzOba>
> .
>